File: modules/calmmama-dev-menu/android/src/main/java/expo/modules/calmmamadevmenu/CalmmamaDevMenuModule.kt
package expo.modules.calmmamadevmenu

import android.app.Application
import expo.modules.devmenu.DevMenuDefaultPreferences
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class CalmmamaDevMenuModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("CalmmamaDevMenu")

    Function("isToolsFabControlAvailable") {
      appContext.reactContext?.applicationContext is Application
    }

    AsyncFunction("setToolsFabVisible") { visible: Boolean ->
      val application = appContext.reactContext?.applicationContext as? Application
        ?: return@AsyncFunction false
      val prefs = DevMenuDefaultPreferences(application)
      prefs.showFab = visible
      true
    }
  }
}

File: modules/calmmama-dev-menu/ios/CalmmamaDevMenuModule.swift
import ExpoModulesCore

#if canImport(EXDevMenu)
import EXDevMenu
#endif

public class CalmmamaDevMenuModule: Module {
  public func definition() -> ModuleDefinition {
    Name("CalmmamaDevMenu")

    Function("isToolsFabControlAvailable") { () -> Bool in
      #if canImport(EXDevMenu) && !os(macOS) && !os(tvOS)
      return true
      #else
      return false
      #endif
    }

    AsyncFunction("setToolsFabVisible") { (visible: Bool) -> Bool in
      #if canImport(EXDevMenu) && !os(macOS) && !os(tvOS)
      DevMenuManager.shared.setShowFloatingActionButton(visible)
      return true
      #else
      return false
      #endif
    }
  }
}
